android/src/main/java/co/apptailor/googlesignin/RNGoogleSigninModule.java

```diff
@@ -258,17 +259,36 @@ public void onResult(@NonNull Status status) {
         });
     }
 
-    @ReactMethod
-    public void getAccessToken(ReadableMap user, Promise promise) {
-        Account acct = new Account(user.getString("email"), "com.google");
-
-        try {
-            String token = GoogleAuthUtil.getToken(getReactApplicationContext(), acct, scopesToString(user.getArray("scopes")));
-            promise.resolve(token);
-        } catch (IOException | GoogleAuthException e) {
-            Log.e(MODULE_NAME, e.getLocalizedMessage());
-            promise.reject(e);
-        }
+    private void getAccessToken(final @NonNull GoogleSignInAccount account) {
```

Comment:
I'd like to ask you to try to improve the method naming and code structure. `getAccessToken` sounds like it returns an access token but it actually returns void and it does a lot of other things - please make it so that it returns just the access token / null if there is a problem. The result can than be added to what is returned from the `getUserProperties(account)` call. Also, starting a new thread to do this seems like an overkill

Reply:
Threads because fo this:
```
     Caused by: java.lang.IllegalStateException: Calling this from your main thread can lead to deadlock
        ...
        at com.google.android.gms.auth.GoogleAuthUtil.getToken(Unknown Source)
```
You're right about naming as in Java get... is a getter and should return something.
I hope you agree then that promise ```resolve``` should be where ```reject``` is (handleSignInResult).
I put getting access token code into ```handleSignInResult``` but probably you want it as a separated method and should I use a callback to return token? What's your opinion?